frontend/src/components/table/TableCell.js

```diff
@@ -259,35 +219,16 @@ class TableCell extends PureComponent {
       viewId,
       modalVisible,
       onClickOutside,
-      isGerman,
     } = this.props;
     const widgetData = getWidgetData(item, isEditable, supportFieldEdit);
-
     const docId = `${this.props.docId}`;
     const { tooltipToggled } = this.state;
-    const tdValue = !isEdited
-      ? TableCell.fieldValueToString(
-          widgetData[0].value,
-          item.widgetType,
-          widgetData[0].precision,
-          isGerman
-        )
-      : null;
-    const description =
-      widgetData[0].value && widgetData[0].value.description
-        ? widgetData[0].value.description
-        : tdValue;
-    let tdTitle =
-      item.widgetType === 'YesNo' ||
-      item.widgetType === 'Switch' ||
-      item.widgetType === 'Color'
-        ? ''
-        : description;
+    const tdValue = this.getTdValue(widgetData);
+    const description = this.getDescription({ widgetData, tdValue });
+    let tdTitle = this.getTdTitle({ item, description });
     const isOpenDatePicker = isEdited && item.widgetType === 'Date';
-    const isDateField = DATE_FIELD_FORMATS[item.widgetType]
-      ? TableCell.getDateFormat(item.widgetType)
-      : false;
-    let style = {};
+    const isDateField = this.checkIfDateField({ item });
```

Comment:
pls, move this line near the place where it is first used.

Also, not sure if it would bring any benefit, but this value can be calculated once on construction time instead of each time on render().

Reply:
As discussed. Props change